pkg/loglines/dedup_test.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package loglines_test
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/loglines"
+)
+
+//nolint:unparam // its only one param for now but we might want more later
+func loadLinesFromFile(path string, generation uint32) (*loglines.LineSlice, error) {
+	reader, err := os.Open(path)
+	if err != nil {
+		return &loglines.LineSlice{}, fmt.Errorf("failed to open file %s %w", path, err)
+	}
+	defer reader.Close()
+	scanner := bufio.NewScanner(reader)
+
+	lines := make([]*loglines.ProcessedLine, 0)
+	for scanner.Scan() {
+		if err := scanner.Err(); err != nil {
+			return &loglines.LineSlice{}, fmt.Errorf("failed to read line from %s %w", path, err)
+		}
+		line, err := loglines.ProcessLine(scanner.Text())
+		if err != nil {
+			return &loglines.LineSlice{}, fmt.Errorf("failed to process line from %s %w", path, err)
+		}
+		lines = append(lines, line)
+	}
+	return loglines.MakeSliceFromLines(lines, generation), nil
+}
+
+var _ = Describe("Dedup AB tests", func() {
```

Comment:
the text in each of the `When` and `It` calls needs to be rewritten for clarity

Reply:
I've gone through them if you think there needs extra let me know.